### ISSUE TITLE
Installing Openfoam package on Arjuna require sudo access

### ISSUE THREAD

**Issue by fujah06** [ArjunaCluster/ArjunaUsers#205]
### Your Name

Olabode

### Andrew ID

oajenifu

### Where it Happened

installation issue

### What Happened?

I am trying to install OpenFOAM software on Arjuna, and the installing involved using sudo and sourcing in bashrc, I have the instruction in the link attached below
https://www.cemf.ir/how-to-install-openfoam-v2012-from-source-pack-on-ubuntu/

I have been able to carry step 1-3, but I could not carry out step 4 and the rest because it requires sudo and sourcing bashrc. I will appreciate if you could guide or give access so that I can install the software. I currently have it on my local computer, but I need it on Arjuna.

Thanks,
Olabode.

**Comment by awadell1** [comment]
Definitely don't open multiple issues for the same issue☹️

But continuing from #187

See our docs for info about installing software on Arjuna: https://arjunacluster.github.io/ArjunaUsers/about/user_software.html

Tldr; we do not provide sudo access to users; additional software beyond what’s provided via modules must be installed via spack/from source.

`spack install openfoam` -- run on a worker node (not the head node!!!) should do it for you

**Comment by fujah06** [comment]
I attempted installing OpenFOAM, but I got this error:
I also referred to this link online
https://developer.amd.com/spack/hpc-applications-openfoam/



Traceback (most recent call last):
  File "/home/spack/spack/lib/spack/spack/main.py", line 1000, in main
    return _main(argv)
  File "/home/spack/spack/lib/spack/spack/main.py", line 955, in _main
    return finish_parse_and_run(parser, cmd_name, env_format_error)
  File "/home/spack/spack/lib/spack/spack/main.py", line 983, in finish_parse_and_run
    return _invoke_command(command, parser, args, unknown)
  File "/home/spack/spack/lib/spack/spack/main.py", line 632, in _invoke_command
    return_val = command(parser, args)
  File "/home/spack/spack/lib/spack/spack/cmd/install.py", line 549, in install
    install_specs(zip(abstract_specs, concrete_specs), install_kwargs, args)
  File "/home/spack/spack/lib/spack/spack/cmd/install.py", line 270, in install_specs
    install_specs_outside_environment(specs, install_kwargs)
  File "/home/spack/spack/lib/spack/spack/cmd/install.py", line 346, in install_specs_outside_environment
    builder.install()
  File "/home/spack/spack/lib/spack/spack/installer.py", line 1858, in install
    pkg=pkg,
spack.installer.InstallError: Installation request failed.  Refer to reported errors for failing package(s).

**Comment by awadell1** [comment]
1) always provide the commands you ran and their entire output. I’m can’t really help if I don’t know what I’m looking at. 2) the docs you linked use the amd compiler. All the nodes on Arjuna are intel and we don’t provide the aocc compiler (you could build it with spack).3) did you try `spack install openfoam`?Best,AlexOn Nov 24, 2022, at 1:00 PM, fujah06 ***@***.***> wrote:﻿
I attempted installing OpenFOAM, but I got this error:
I also referred to this link online
https://developer.amd.com/spack/hpc-applications-openfoam/
Traceback (most recent call last):
File "/home/spack/spack/lib/spack/spack/main.py", line 1000, in main
return _main(argv)
File "/home/spack/spack/lib/spack/spack/main.py", line 955, in _main
return finish_parse_and_run(parser, cmd_name, env_format_error)
File "/home/spack/spack/lib/spack/spack/main.py", line 983, in finish_parse_and_run
return _invoke_command(command, parser, args, unknown)
File "/home/spack/spack/lib/spack/spack/main.py", line 632, in _invoke_command
return_val = command(parser, args)
File "/home/spack/spack/lib/spack/spack/cmd/install.py", line 549, in install
install_specs(zip(abstract_specs, concrete_specs), install_kwargs, args)
File "/home/spack/spack/lib/spack/spack/cmd/install.py", line 270, in install_specs
install_specs_outside_environment(specs, install_kwargs)
File "/home/spack/spack/lib/spack/spack/cmd/install.py", line 346, in install_specs_outside_environment
builder.install()
File "/home/spack/spack/lib/spack/spack/installer.py", line 1858, in install
pkg=pkg,
spack.installer.InstallError: Installation request failed.  Refer to reported errors for failing package(s).

**Comment by fujah06** [comment]
Hi 
One of the question is how will I access worker node? also the version of openFOAM I am try to install is OpenFOAM-v2012.
I input this code below
[oajenifu@coe OpenFOAM]$ spack install openfoam

[oajenifu@coe OpenFOAM]$ spack install openfoam
[+] /usr (external diffutils-3.3-ddwqy7zilbfgwnc7n5dmcl5ku2fyiyot)
[+] /usr/local (external cmake-3.17.5-26afwzswaq66qs4ftmq366eji3alp6sg)
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/lz4-1.9.4-23uckchtqrsxovlh7oltizb4gn5dqnvc
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/zlib-1.2.11-xzfuiy6a5csxs6l6rpezodmkv6hjtbm3
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/zstd-1.5.2-25nqizaz3a5kirhohatiseicpcgosp5f
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/libfabric-1.12.1-ykqsqjtiqqd23vaz625i7xnoa5ydp4s6
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/libffi-3.4.2-sd3rout25hkytum2eu76hfbgngemco46
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/nvhpc-21.7-xhrq4yufsdarynoudqwsn67qmmk4e4yj
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/pkgconf-1.8.0-tlpr63jljqhlyjiy4nwip3ic5ygg26qr
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/xz-5.2.5-mv6ce3ep66vmbelf52swnflgln7qeguy
[+] /usr (external autoconf-2.69-f6rrnlx3hvsurbtncbinjqjwfzl5wlia)
[+] /usr (external automake-1.13.4-awhncolns7acsa4uishm2x4gkclmqku3)
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/libtool-2.4.6-dgfudc4uqitcp5p54gzbdzskl6vnakr2
[+] /usr (external m4-1.4.16-2bkgwzokyebzoc3xi2ftkru4quzf3dkp)
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/autoconf-archive-2022.02.11-anj72mjj7acfedn3qqlxdn24aj3ohhss
[+] /usr (external texinfo-5.1-kvsj6fmnja65yjm4lwfbfybv74hyg56y)
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/berkeley-db-18.1.40-raq5fvz2lrnul3oxtq2mdetr4koouu5s
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/ncurses-6.2-lr4ubvzrejsumoyushv6qg2wdbihhzmm
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/findutils-4.8.0-4mmsk2goryoe4wspx3m7od5oa77a3b4s
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/libiconv-1.16-n33ct7hnqgykzmecqzmwc3jdnchekya5
[+] /usr (external pkg-config-0.27.1-edmt2kgnjlm35yz7myrwhmq4oonwkzxv)
[+] /usr (external tar-1.26-lplcned3ksrqdddmsp4se5iscc3p6enj)
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/bzip2-1.0.8-lvqvb5ljcqr4pprc6jlcjm7hvnifxaut
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/zfp-0.5.5-erwipbeaxwxrpmivnumyfc3laxdjcpgh
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/eigen-3.4.0-uz2ezlwyikfvgfvhlokjaheax5n6jjml
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/snappy-1.1.9-vwkdbda4zspbuiwgdsi373er3asrdnm3
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/libpng-1.6.37-3crx2m7f2r6hagxg7o3ntxjxx4lloarn
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/sz-2.1.12.2-yh7wwdgjxncrqo63t6efax4lcnb4fxyy
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/fftw-3.3.10-gy5tbzy7npnoq6ujvggceypt35vbqy6e
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/gmp-6.2.1-zoppkslhkcx5femdg72skwlfjylrdnqt
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/readline-8.1-sq7o5mxbzjjcmqoz3xeqhmoi6f3usxrm
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/libxml2-2.9.13-6yoyet44nu7zky7lxsfxvbed2manqpf7
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/boost-1.80.0-7xxev6x7udxcxjymejgxjirhlqz7p5na
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/c-blosc-1.21.1-cdi54b7ekucn3zvjfehrj2q3k5ezud7j
==> Installing mpfr-4.1.0-r6uq6mruisxc3qclzavux2ms6n5nrauf
==> No binary for mpfr-4.1.0-r6uq6mruisxc3qclzavux2ms6n5nrauf found: installing from source
==> Using cached archive: /home/oajenifu/.spack/var/spack/cache/_source-cache/archive/fe/feced2d430dd5a97805fa289fed3fc8ff2b094c02d05287fd6133e7f1f0ec926.tar.bz2
==> No patches needed for mpfr
==> mpfr: Executing phase: 'autoreconf'
==> mpfr: Executing phase: 'configure'
==> mpfr: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16' 'V=1'

5 errors found in build log:
     18     libtoolize: copying file 'm4/ltsugar.m4'
     19     libtoolize: copying file 'm4/ltversion.m4'
     20     libtoolize: copying file 'm4/lt~obsolete.m4'
     21     autoreconf: running: /usr/bin/autoconf --include=/home/oajenifu/.sp
            ack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/autoconf-archive-2
            022.02.11-anj72mjj7acfedn3qqlxdn24aj3ohhss/share/aclocal --include=
            /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/lib
            tool-2.4.6-dgfudc4uqitcp5p54gzbdzskl6vnakr2/share/aclocal --force
     22     autoreconf: configure.ac: not using Autoheader
     23     autoreconf: running: automake --add-missing --copy --force-missing
  >> 24     configure.ac:481: installing './compile'
  >> 25     configure.ac:37: installing './missing'
     26     src/Makefile.am: installing './depcomp'
     27     autoreconf: Leaving directory `.'
     28     ==> mpfr: Executing phase: 'configure'
     29     ==> [2022-11-24-16:25:31.573088] FILTER FILE: /tmp/oajenifu/spack-s
            tage/spack-stage-mpfr-4.1.0-r6uq6mruisxc3qclzavux2ms6n5nrauf/spack-
            src/configure [replacing "^(\s*if test x-L = )("\$p" \|\|\s*)$"]
     30     ==> [2022-11-24-16:25:31.644696] FILTER FILE: /tmp/oajenifu/spack-s
            tage/spack-stage-mpfr-4.1.0-r6uq6mruisxc3qclzavux2ms6n5nrauf/spack-
            src/configure [replacing "^(\s*test x-R = )("\$p")(; then\s*)$"]
     31     ==> [2022-11-24-16:25:31.696076] FILTER FILE: /tmp/oajenifu/spack-s
            tage/spack-stage-mpfr-4.1.0-r6uq6mruisxc3qclzavux2ms6n5nrauf/spack-
            src/configure [replacing "^(\s*test \$p = "-R")(; then\s*)$"]

     ...

     240      rc=$?; \
     241      CDPATH="${ZSH_VERSION+.}:" && cd . && \
     242      $restore $backupdir/* `echo "./mpfr.info" | sed 's|[^/]*$||'`; \
     243    fi; \
     244    rm -rf $backupdir; exit $rc
     245    /tmp/oajenifu/spack-stage/spack-stage-mpfr-4.1.0-r6uq6mruisxc3qclza
            vux2ms6n5nrauf/spack-src/missing: line 81: makeinfo: command not fo
            und
  >> 246    WARNING: 'makeinfo' is missing on your system.
     247             You should only need it if you modified a '.texi' file, or
     248             any other file indirectly affecting the aspect of the manu
            al.
     249             You might want to install the Texinfo package:
     250             <http://www.gnu.org/software/texinfo/>
     251             The spurious makeinfo call might also be the consequence o
            f
     252             using a buggy 'make' (AIX, DU, IRIX), in which case you mi
            ght
     253             want to install GNU make:
     254             <http://www.gnu.org/software/make/>
  >> 255    make[1]: *** [mpfr.info] Error 127
     256    make[1]: Leaving directory `/tmp/oajenifu/spack-stage/spack-stage-m
            pfr-4.1.0-r6uq6mruisxc3qclzavux2ms6n5nrauf/spack-src/doc'
  >> 257    make: *** [all-recursive] Error 1

See build log for details:
  /tmp/oajenifu/spack-stage/spack-stage-mpfr-4.1.0-r6uq6mruisxc3qclzavux2ms6n5nrauf/spack-build-out.txt

==> Warning: Skipping build of cgal-4.13-eckowjaflwnv4kwjtyampmzlqkayolf4 since mpfr-4.1.0-r6uq6mruisxc3qclzavux2ms6n5nrauf failed
==> Warning: Skipping build of openfoam-2206-idmnj4le7zffmv6tmzvfyatzqv3bsx3s since cgal-4.13-eckowjaflwnv4kwjtyampmzlqkayolf4 failed
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/gdbm-1.19-jdiy5izajukay6p5ceoj23eogi77phvv
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/gettext-0.21-r3z4bgu7rijv25qj4qutgljgo6u4kgvc
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/adios2-2.8.3-ugbm2b3wtwm5ynjmyo3xrbuavb7inrd5
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/perl-5.34.1-kcswftfwbbpp5v2gimfwflhafny35ftu
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/help2man-1.47.16-brddcvludfs67kgm2cehjqsaksban7zr
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/bison-3.8.2-olc22e7r4iw4y4gjl5gomnusm5tfb3y4
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/flex-2.6.4-bjskelhc7rdntkhftnbynf72rzetq55c
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/scotch-7.0.1-i2tdzd7bzvafzdir7wjdeazz7ej3gxqy
==> Error: openfoam-2206-idmnj4le7zffmv6tmzvfyatzqv3bsx3s: Package was not installed
==> Error: Installation request failed.  Refer to reported errors for failing package(s).
[oajenifu@coe OpenFOAM]$

**Comment by fujah06** [comment]
Hi, following up with my previous message.
One of the questions is how will I access worker node? also the version of openFOAM I am try to install is OpenFOAM-v2012.
I input this code below
[oajenifu@coe OpenFOAM]$ spack install openfoam

[oajenifu@coe OpenFOAM]$ spack install openfoam
[+] /usr (external diffutils-3.3-ddwqy7zilbfgwnc7n5dmcl5ku2fyiyot)
[+] /usr/local (external cmake-3.17.5-26afwzswaq66qs4ftmq366eji3alp6sg)
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/lz4-1.9.4-23uckchtqrsxovlh7oltizb4gn5dqnvc
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/zlib-1.2.11-xzfuiy6a5csxs6l6rpezodmkv6hjtbm3
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/zstd-1.5.2-25nqizaz3a5kirhohatiseicpcgosp5f
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/libfabric-1.12.1-ykqsqjtiqqd23vaz625i7xnoa5ydp4s6
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/libffi-3.4.2-sd3rout25hkytum2eu76hfbgngemco46
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/nvhpc-21.7-xhrq4yufsdarynoudqwsn67qmmk4e4yj
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/pkgconf-1.8.0-tlpr63jljqhlyjiy4nwip3ic5ygg26qr
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/xz-5.2.5-mv6ce3ep66vmbelf52swnflgln7qeguy
[+] /usr (external autoconf-2.69-f6rrnlx3hvsurbtncbinjqjwfzl5wlia)
[+] /usr (external automake-1.13.4-awhncolns7acsa4uishm2x4gkclmqku3)
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/libtool-2.4.6-dgfudc4uqitcp5p54gzbdzskl6vnakr2
[+] /usr (external m4-1.4.16-2bkgwzokyebzoc3xi2ftkru4quzf3dkp)
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/autoconf-archive-2022.02.11-anj72mjj7acfedn3qqlxdn24aj3ohhss
[+] /usr (external texinfo-5.1-kvsj6fmnja65yjm4lwfbfybv74hyg56y)
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/berkeley-db-18.1.40-raq5fvz2lrnul3oxtq2mdetr4koouu5s
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/ncurses-6.2-lr4ubvzrejsumoyushv6qg2wdbihhzmm
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/findutils-4.8.0-4mmsk2goryoe4wspx3m7od5oa77a3b4s
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/libiconv-1.16-n33ct7hnqgykzmecqzmwc3jdnchekya5
[+] /usr (external pkg-config-0.27.1-edmt2kgnjlm35yz7myrwhmq4oonwkzxv)
[+] /usr (external tar-1.26-lplcned3ksrqdddmsp4se5iscc3p6enj)
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/bzip2-1.0.8-lvqvb5ljcqr4pprc6jlcjm7hvnifxaut
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/zfp-0.5.5-erwipbeaxwxrpmivnumyfc3laxdjcpgh
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/eigen-3.4.0-uz2ezlwyikfvgfvhlokjaheax5n6jjml
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/snappy-1.1.9-vwkdbda4zspbuiwgdsi373er3asrdnm3
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/libpng-1.6.37-3crx2m7f2r6hagxg7o3ntxjxx4lloarn
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/sz-2.1.12.2-yh7wwdgjxncrqo63t6efax4lcnb4fxyy
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/fftw-3.3.10-gy5tbzy7npnoq6ujvggceypt35vbqy6e
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/gmp-6.2.1-zoppkslhkcx5femdg72skwlfjylrdnqt
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/readline-8.1-sq7o5mxbzjjcmqoz3xeqhmoi6f3usxrm
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/libxml2-2.9.13-6yoyet44nu7zky7lxsfxvbed2manqpf7
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/boost-1.80.0-7xxev6x7udxcxjymejgxjirhlqz7p5na
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/c-blosc-1.21.1-cdi54b7ekucn3zvjfehrj2q3k5ezud7j
==> Installing mpfr-4.1.0-r6uq6mruisxc3qclzavux2ms6n5nrauf
==> No binary for mpfr-4.1.0-r6uq6mruisxc3qclzavux2ms6n5nrauf found: installing from source
==> Using cached archive: /home/oajenifu/.spack/var/spack/cache/_source-cache/archive/fe/feced2d430dd5a97805fa289fed3fc8ff2b094c02d05287fd6133e7f1f0ec926.tar.bz2
==> No patches needed for mpfr
==> mpfr: Executing phase: 'autoreconf'
==> mpfr: Executing phase: 'configure'
==> mpfr: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
'make' '-j16' 'V=1'

5 errors found in build log:
18 libtoolize: copying file 'm4/ltsugar.m4'
19 libtoolize: copying file 'm4/ltversion.m4'
20 libtoolize: copying file 'm4/lt~obsolete.m4'
21 autoreconf: running: /usr/bin/autoconf --include=/home/oajenifu/.sp
ack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/autoconf-archive-2
022.02.11-anj72mjj7acfedn3qqlxdn24aj3ohhss/share/aclocal --include=
/home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/lib
tool-2.4.6-dgfudc4uqitcp5p54gzbdzskl6vnakr2/share/aclocal --force
22 autoreconf: configure.ac: not using Autoheader
23 autoreconf: running: automake --add-missing --copy --force-missing

24 configure.ac:481: installing './compile'
25 configure.ac:37: installing './missing'
26 src/Makefile.am: installing './depcomp'
27 autoreconf: Leaving directory `.'
28 ==> mpfr: Executing phase: 'configure'
29 ==> [2022-11-24-16:25:31.573088] FILTER FILE: /tmp/oajenifu/spack-s
tage/spack-stage-mpfr-4.1.0-r6uq6mruisxc3qclzavux2ms6n5nrauf/spack-
src/configure [replacing "^(\sif test x-L = )("$p" ||\s)$"]
30 ==> [2022-11-24-16:25:31.644696] FILTER FILE: /tmp/oajenifu/spack-s
tage/spack-stage-mpfr-4.1.0-r6uq6mruisxc3qclzavux2ms6n5nrauf/spack-
src/configure [replacing "^(\stest x-R = )("$p")(; then\s)$"]
31 ==> [2022-11-24-16:25:31.696076] FILTER FILE: /tmp/oajenifu/spack-s
tage/spack-stage-mpfr-4.1.0-r6uq6mruisxc3qclzavux2ms6n5nrauf/spack-
src/configure [replacing "^(\stest $p = "-R")(; then\s)$"]

 ...

 240      rc=$?; \
 241      CDPATH="${ZSH_VERSION+.}:" && cd . && \
 242      $restore $backupdir/* `echo "./mpfr.info" | sed 's|[^/]*$||'`; \
 243    fi; \
 244    rm -rf $backupdir; exit $rc
 245    /tmp/oajenifu/spack-stage/spack-stage-mpfr-4.1.0-r6uq6mruisxc3qclza
        vux2ms6n5nrauf/spack-src/missing: line 81: makeinfo: command not fo
        und
246 WARNING: 'makeinfo' is missing on your system.
247 You should only need it if you modified a '.texi' file, or
248 any other file indirectly affecting the aspect of the manu
al.
249 You might want to install the Texinfo package:
250 http://www.gnu.org/software/texinfo/
251 The spurious makeinfo call might also be the consequence o
f
252 using a buggy 'make' (AIX, DU, IRIX), in which case you mi
ght
253 want to install GNU make:
254 http://www.gnu.org/software/make/
255 make[1]: *** [mpfr.info] Error 127
256 make[1]: Leaving directory `/tmp/oajenifu/spack-stage/spack-stage-m
pfr-4.1.0-r6uq6mruisxc3qclzavux2ms6n5nrauf/spack-src/doc'
257 make: *** [all-recursive] Error 1

See build log for details:
/tmp/oajenifu/spack-stage/spack-stage-mpfr-4.1.0-r6uq6mruisxc3qclzavux2ms6n5nrauf/spack-build-out.txt

==> Warning: Skipping build of cgal-4.13-eckowjaflwnv4kwjtyampmzlqkayolf4 since mpfr-4.1.0-r6uq6mruisxc3qclzavux2ms6n5nrauf failed
==> Warning: Skipping build of openfoam-2206-idmnj4le7zffmv6tmzvfyatzqv3bsx3s since cgal-4.13-eckowjaflwnv4kwjtyampmzlqkayolf4 failed
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/gdbm-1.19-jdiy5izajukay6p5ceoj23eogi77phvv
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/gettext-0.21-r3z4bgu7rijv25qj4qutgljgo6u4kgvc
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/adios2-2.8.3-ugbm2b3wtwm5ynjmyo3xrbuavb7inrd5
[+] /home/spack/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/perl-5.34.1-kcswftfwbbpp5v2gimfwflhafny35ftu
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/help2man-1.47.16-brddcvludfs67kgm2cehjqsaksban7zr
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/bison-3.8.2-olc22e7r4iw4y4gjl5gomnusm5tfb3y4
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/flex-2.6.4-bjskelhc7rdntkhftnbynf72rzetq55c
[+] /home/oajenifu/.spack/opt/spack/linux-centos7-broadwell/gcc-11.2.0/scotch-7.0.1-i2tdzd7bzvafzdir7wjdeazz7ej3gxqy
==> Error: openfoam-2206-idmnj4le7zffmv6tmzvfyatzqv3bsx3s: Package was not installed
==> Error: Installation request failed. Refer to reported errors for failing package(s).
[oajenifu@coe OpenFOAM]$

**Comment by awadell1** [comment]
I’d encourage you to check out the [getting started](https://arjunacluster.github.io/ArjunaUsers/getting_started/) section of our documentation. Specifically, the sections on slurm and submitting your first job.

It looks like the build is failing on mpfr. As we do have GNU make installed, I'd consult their documentation for possible issues.

**Comment by fujah06** [comment]
I have done the reading as you suggested. can we have a zoom call to discuss? I found some resources which may help in creating the working node file at 
https://www.cfd-online.com/Forums/openfoam-solving/183018-slurm-job-submission-steps.html

Typically I have a script that I use to run the job on my local computer. 

#!/bin/sh
cd "${0%/*}" || exit                                # Run from this directory
. ${WM_PROJECT_DIR:?}/bin/tools/RunFunctions        # Tutorial run functions
#------------------------------------------------------------------------------
touch slm.foam
restore0Dir

runApplication blockMesh
//checkMesh > log.checkMesh

runApplication decomposePar

echo -e "\033[33mrenumberMesh\033[0m"
time mpirun --use-hwthread-cpus -np 4 renumberMesh -parallel -overwrite > log.renumberMesh

echo -e "\033[33msetFields\033[0m"
mpirun -np 4 setFields -parallel > log.setFields
//runApplication setFields

runParallel `getApplication`
//runApplication $(getApplication)

**Comment by amalss18** [comment]
@CoopLo @zcao0420 Could you help @fujah06 with using slurm.
